fog/view/server/src/router_request_handler.rs

```diff
@@ -105,7 +102,7 @@ async fn handle_query_request<E>(
 where
     E: ViewEnclaveProxy,
 {
-    let mut query_responses: BTreeMap<ResponderId, EnclaveMessage<NonceSession>> = BTreeMap::new();
+    let mut query_responses: BTreeMap<ResponderId, MultiViewStoreQueryResponse> = BTreeMap::new();
```

Comment:
I must be missing it.
- on [150](https://github.com/mobilecoinfoundation/mobilecoin/blob/sam/mvq-internal/fog/view/server/src/router_request_handler.rs#L150) the [`ProcessedShardResponseData::multi_view_store_query_responses`](https://github.com/mobilecoinfoundation/mobilecoin/blob/sam/mvq-internal/fog/view/server/src/shard_responses_processor.rs#L22) are added to the `query_responses`
- in [`process_shard_responses`](https://github.com/mobilecoinfoundation/mobilecoin/blob/sam/mvq-internal/fog/view/server/src/shard_responses_processor.rs#L50) it looks like only `Success` queries are added

It looks like authentication errors get sent to [`view_store_uris_for_authentication`](https://github.com/mobilecoinfoundation/mobilecoin/blob/sam/mvq-internal/fog/view/server/src/shard_responses_processor.rs#L58)

Reply:
Oh yes, that's right. I just re-looked at it, and I remembered that we need the `ResponderId` to be attached to the encrypted query response because the ake enclave needs the responder id to pick the right decryptor. THerefore, we could use a `Vec<(ResponderId, attest::NonceMessage)>` but we could also use a `BTreeMap`. I don't see much of an advantage to the Vec solution, whereas the BTreeMap enforces no duplicates. Make sense?

Reply:
Sorry, I'm still missing it.
I only see the `query_response.into_values()` being used, which turns into a vec of `MultiViewStoreQueryResponse`

Are you saying it's coming later?

Reply:
This patch builds fine, [vector.txt](https://github.com/mobilecoinfoundation/mobilecoin/files/9815318/vector.txt)

Reply:
Oh I see it now (was looking at an old branch). Done.